### PR TITLE
agent: Extract and improve NixOS snippets

### PR DESCRIPTION
**Pull request by roberth**

### CODE REVIEW

File: docs/modules/ROOT/partials/snippets/NixOSToken.adoc [hunk 1]
```diff
@@ -0,0 +1 @@
+include::partial$generate-token.adoc[]
```
Comment (domenkozar) [inline docs/modules/ROOT/partials/snippets/NixOSToken.adoc]: I don't see a good reason to create another indirection?

Comment (roberth) [reply docs/modules/ROOT/partials/snippets/NixOSToken.adoc]: This makes the snippet available to the frontend.

Comment (domenkozar) [reply docs/modules/ROOT/partials/snippets/NixOSToken.adoc]: Because of camelcase naming? So why not rename the original?

Comment (roberth) [reply docs/modules/ROOT/partials/snippets/NixOSToken.adoc]: Not just that, partial/generate-token.adoc is not in the snippet directory.
If you think partial$generate-token.adoc needs to be inlined, we can refactor that separately. I'm merely following the pattern that was already there, to achieve a goal of fixing the frontend.

**Comment by roberth** [comment]
Created https://github.com/hercules-ci/docs.hercules-ci.com/issues/86

**Pull request merged**